### PR TITLE
HPCC-14382 LDAP build break on Windows

### PR DESCRIPTION
**Pull request by RussWhitehead**
WinLDAP defines ldap_init differently than the OpenLDAP call ldap_initialize,
which causes a platform build break in Windows. This PR corrects that problem,
and adds the export of the verifyServerCert method

Signed-off-by: Russ Whitehead william.whitehead@lexisnexis.com

### CODE REVIEW

**Comment by RussWhitehead** [comment]
@ghalliday I had to create this new pull request. Please review

File: system/security/LdapSecurity/ldaputils.cpp [hunk 1]
```diff
@@ -81,14 +80,22 @@ LDAP* LdapUtils::LdapInit(const char* protocol, const char* host, int port, int
...
+        {
+            throw MakeStringException(-1, "ldap_init(%s,%d) error %s", host, port, ldap_err2string(LdapGetLastError()));
+        }
+#else
+        StringBuffer uri("ldap://");
+        uri.appendf("%s:%d", host, port);
         int rc = LDAP_INIT(&ld, uri.str());
         if(rc != LDAP_SUCCESS)
         {
-            throw MakeStringException(-1, "ldap_initialize(%s) error %s", uri.str(), ldap_err2string(rc));
```
Comment (ghalliday) [inline system/security/LdapSecurity/ldaputils.cpp]: trivial: Curious why this was changed.

File: system/security/LdapSecurity/ldaputils.cpp [hunk 2]
```diff
@@ -71,7 +71,6 @@ LDAP* LdapUtils::LdapInit(const char* protocol, const char* host, int port, int
         int rc = LDAP_INIT(&ld, uri.str());
         if(rc != LDAP_SUCCESS)
         {
-            DBGLOG("ldap_initialize error %s", ldap_err2string(rc));
```
Comment (ghalliday) [inline system/security/LdapSecurity/ldaputils.cpp]: trivial: Why was this deleted?

**Comment by ghalliday** [comment]
@RussWhitehead I would be happy to merge as it is, but had a couple of minor questions before I do.

File: system/security/LdapSecurity/ldaputils.cpp (hunk 2 above)
Comment (RussWhitehead) [inline system/security/LdapSecurity/ldaputils.cpp]: Throwing the string exception logs the error, so the DBGLOG seems unnecessary

File: system/security/LdapSecurity/ldaputils.cpp (hunk 1 above)
Comment (RussWhitehead) [inline system/security/LdapSecurity/ldaputils.cpp]: I added it back

**Comment by RussWhitehead** [comment]
@ghalliday Please rereview

**Comment by RussWhitehead** [comment]
@ghalliday Fixed smoke test build break. Please review
